src/modules/ocaml.rs

```diff
@@ -29,6 +30,18 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
                 _ => None,
             })
             .map(|variable| match variable {
+                "switch" => {
+                    let stdout = context.exec_cmd("opam", &["switch", "-s", "show"])?.stdout;
+                    let switch_raw = stdout.trim();
+                    let path = Path::new(switch_raw);
+                    let normalized = if path.has_root() {
+                        // use named local switch, oft prefixed with full path by opam
+                        path.file_name()?.to_str()?
+                    } else {
+                        switch_raw
+                    };
```

Comment:
I may be wrong about this but couldn't you just take the filename portion unconditionally? 🤔

Reply:
switches are absolute paths, or not path based. this check protects against mangling a non-path based switch name

Reply:
I understand your motivation but I don't think there is a way to mangle named switch by extracting filename from it. It seems that OPAM authors agree with me: https://github.com/ocaml/opam/blob/9f9e9297/doc/pages/Tricks.md#display-the-current-opam-switch-in-the-prompt-

Reply:
ya, you're right. i did some tests locally, and i think it's all zen. updated in 279adb6